### PR TITLE
feat(travis): add JDK10 and JDK11 tests

### PR DESCRIPTION
**Pull request by maxhbr**
This PR also adds CLI tests

### CODE REVIEW

**Comment by maxhbr** [comment]
This PR currently adds the explicit dependency to the package
```xml
<dependency>
    <groupId>javax.xml.bind</groupId>
    <artifactId>jaxb-api</artifactId>
    <version>2.2.11</version>
</dependency>
```
is licensed under `CDDL-1.1 OR GPL-2.0-w/-CPE`.

File: .travis.yml [hunk 1]
```diff
@@ -6,30 +6,81 @@
...
-  # - openjdk10
-  # - openjdk11
-  # - oraclejdk8
-  # - oraclejdk9
-  # - oraclejdk11
+addons:
+  apt:
+    packages:
+      - git
+# cache:
```
Comment (greimela-si) [inline .travis.yml]: Should we enable caching again?

File: .travis.yml [hunk 2]
```diff
@@ -6,30 +6,81 @@
...
-      install:
-        - mvn clean verify -P '!build-assembly' -DskipTests=true -V -B
-        - mvn install -P '!build-assembly' -DskipTests=true -B
+    # compile and execute with the same JDK
+    - &mvn-test
+      name: mvn test
+      script:
+        - mvn test -B
+        - cd example-projects/example-project
+        - mvn package
```
Comment (greimela-si) [inline .travis.yml]: Use `mvn clean test` and `mvn clean package` to isolate the different test cases?

File: .travis.yml [hunk 3]
```diff
@@ -6,30 +6,81 @@
...
+    - &differentJDK-test
+      name: mvn test, run with openjdk9
+      env:
+        - EXECUTION_JDK=9
       script:
-        - mvn test -P '!build-assembly' -B
+        - export JAVA_HOME=$HOME/custom_jdk
+        - /tmp/install-jdk.sh -f $EXECUTION_JDK -l GPL --target $JAVA_HOME
+        - cd example-projects/example-project
+        - mvn package -Dmaven.compiler.source=$EXECUTION_JDK -Dmaven.compiler.target=$EXECUTION_JDK
```
Comment (greimela-si) [inline .travis.yml]: Also use `clean` here.

File: .travis.yml (hunk 1 above)
Comment (maxhbr) [reply .travis.yml]: I would enable caching in the next PR, I am also unsure whether it helps. Travis says: if you cache something which you would load from the internet, then loading it from the internet might be faster than loading it from the travis cache.

File: .travis.yml (hunk 2 above)
Comment (maxhbr) [reply .travis.yml]: mvn package is ran in a subdirectory, which has not yet seen any interaction

File: .travis.yml (hunk 3 above)
Comment (maxhbr) [reply .travis.yml]: This is a fresh checked out subdirectory (`example-projects/example-project`) in which no mvn cmd was called before

**Comment by maxhbr** [comment]
@greimela-si can you please review again?

**Pull request merged**